Editor/Scripts/CodeGenerators/BuildAutoProcessor.cs

```diff
@@ -0,0 +1,47 @@
+using System.Linq;
+using RegressionGames;
+#if UNITY_EDITOR
+using RegressionGames.Editor.CodeGenerators;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+namespace RegressionGames.Editor.CodeGenerators
+{
+    class BuildAutoPreProcessor : IPreprocessBuildWithReport
+    {
+        public int callbackOrder { get { return 0; } }
+        
+        public void OnPreprocessBuild(BuildReport report)
+        {
+            BuildAssetPostProcessor.TriggerGeneration();
+        }
+    }
+   
+    class BuildAssetPostProcessor : UnityEditor.AssetPostprocessor
+    {
+
+        static bool _enabled = true;
+        
+        public static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets,
+            string[] movedFromAssetPaths, bool didDomainReload)
+        {
+            // domain reload after build resets enabled to true
+            // and this gets called once with the original script changes, once with the generated changes, and then one more time with an empty list
+            if (_enabled && importedAssets.FirstOrDefault(ia => ia.EndsWith(".cs") && !ia.StartsWith("Packages/gg.regression.unity.bots") && !ia.Contains("RegressionGames/Runtime/GeneratedScripts")) != null)
```

Comment:
I remember that you provided feedback that the concept in this PR caused the generation to happen too often and was a nuisance - is that still an issue? If I am working on random files in my project unrelated to projects, will generation occur?

Reply:
It is far more reasonable with these new filters added.  I did a PoC of scanning the changes file(s) sources for the presence of [RGState] or [RGAction] strings, but ...
- That didn't scale up well with many files changed
- It didn't catch removals

IOW.. for now, this is as good as it gets.  We can filter out more in the future if necessary.

Reply:
I think that makes sense. The App Domain will have to reload any time Unity recompiles the scripts and that will indeed reset `_enabled` to `true`.